### ISSUE TITLE
AWS Local Zone Support

### ISSUE THREAD

**Issue by johnbairmbo** [F5Networks/f5-appsvcs-extension#400]
AWS:
US-WEST-2 with Local Zone's
BIG-IP 15.1.2 Build 0.0.9 Final

### Summary
We are working to migrate our applications to the local zone in LAX.  These are setup as us-west-2-lax-1a or 1b.  We currently have a cloud formation built for 2 nics in accross zone formation.  All things look well with the setup.  However the service discovery will not bring in any nodes into the config.  We have tested with systems tagged both in the local zone and in the parent zone.  

### Steps To Reproduce
1. Submit the following declaration:
```json
{
    "class": "ADC",
    "schemaVersion": "3.24.0",
    "id": "urn:uuid:33045210-3ab8-4636-9b2a-c98d22ab425d",
    "label": "AWS Service Discovery",
    "remark": "Simple HTTP application with a pool using AWS service discovery",
    "ASG": {
      "class": "Tenant",
      "A1": {
        "class": "Application",
        "service": {
          "class": "Service_HTTP",
          "virtualAddresses": [
            "10.213.130.124"          ],
          "pool": "web_pool"
        },
        "web_pool": {
          "class": "Pool",
          "monitors": [
            "http"
          ],
          "members": [       
            {
              "servicePort": 80,
              "addressDiscovery": "aws",
              "updateInterval": 1,
              "tagKey": "foo",
              "tagValue": "bar",
              "addressRealm": "private",
              "region": "us-west-2",
              "credentialUpdate": false
            }
          ]
        }
      }
    }
  }  
```

All of the needed things get built but the nodes never show up.
```

### Expected Behavior
Expecting the nodes to show up to the system and be members of the pool as defined.

### Actual Behavior
The partition, virtual server and pool all get created but no nodes.

**Comment by dstokesf5** [comment]
Changing this to a feature request since AWS Local Zone support will need to be added to F5Networks/f5-cloud-libs-aws.

**Comment by dstokesf5** [comment]
It looks like Local Zones are mostly tied into the existing APIs that f5-cloud-libs-aws is using. I overlooked the fact that you were not able to discover instance from the parent zone as well. Do you see any errors from service discovery in `/var/log/restnoded/restnoded.log` on the BIG-IP?

**Comment by johnbairmbo** [comment]
We are able to discover instances if we place the F5 itself in the parent zone.  Will go get a copy of the restnoded.log on the BIG-IP.

**Comment by johnbairmbo** [comment]
Tue, 05 Jan 2021 19:16:32 GMT - severe: [f5-service-discovery: getNodes.js] error code: UnauthorizedOperation message: You are not authorized to perform this operation.
Tue, 05 Jan 2021 19:16:32 GMT - info: [f5-service-discovery: Task.js] Task ~ASG~5PxpYvu7uFu5yBVUT6CH6uu2BjbJQUQdeRpcz5AQcx1w3D failed: You are not authorized to perform this operation.
Tue, 05 Jan 2021 19:16:32 GMT - fine: [f5-service-discovery: Task.js] UnauthorizedOperation: You are not authorized to perform this operation.
    at Request.extractError (/var/config/rest/iapps/f5-service-discovery/node_modules/aws-sdk/lib/services/ec2.js:50:35)
    at Request.callListeners (/var/config/rest/iapps/f5-service-discovery/node_modules/aws-sdk/lib/sequential_executor.js:106:20)
    at Request.emit (/var/config/rest/iapps/f5-service-discovery/node_modules/aws-sdk/lib/sequential_executor.js:78:10)
    at Request.emit (/var/config/rest/iapps/f5-service-discovery/node_modules/aws-sdk/lib/request.js:688:14)
    at Request.transition (/var/config/rest/iapps/f5-service-discovery/node_modules/aws-sdk/lib/request.js:22:10)
    at AcceptorStateMachine.runTo (/var/config/rest/iapps/f5-service-discovery/node_modules/aws-sdk/lib/state_machine.js:14:12)
    at /var/config/rest/iapps/f5-service-discovery/node_modules/aws-sdk/lib/state_machine.js:26:10
    at Request.<anonymous> (/var/config/rest/iapps/f5-service-discovery/node_modules/aws-sdk/lib/request.js:38:9)
    at Request.<anonymous> (/var/config/rest/iapps/f5-service-discovery/node_modules/aws-sdk/lib/request.js:690:12)
    at Request.callListeners (/var/config/rest/iapps/f5-service-discovery/node_modules/aws-sdk/lib/sequential_executor.js:116:18)
    at Request.emit (/var/config/rest/iapps/f5-service-discovery/node_modules/aws-sdk/lib/sequential_executor.js:78:10)
    at Request.emit (/var/config/rest/iapps/f5-service-discovery/node_modules/aws-sdk/lib/request.js:688:14)
    at Request.transition (/var/config/rest/iapps/f5-service-discovery/node_modules/aws-sdk/lib/request.js:22:10)
    at AcceptorStateMachine.runTo (/var/config/rest/iapps/f5-service-discovery/node_modules/aws-sdk/lib/state_machine.js:14:12)
    at /var/config/rest/iapps/f5-service-discovery/node_modules/aws-sdk/lib/state_machine.js:26:10
    at Request.<anonymous> (/var/config/rest/iapps/f5-service-discovery/node_modules/aws-sdk/lib/request.js:38:9)
    at Request.<anonymous> (/var/config/rest/iapps/f5-service-discovery/node_modules/aws-sdk/lib/request.js:690:12)
    at Request.callListeners (/var/config/rest/iapps/f5-service-discovery/node_modules/aws-sdk/lib/sequential_executor.js:116:18)
    at callNextListener (/var/config/rest/iapps/f5-service-discovery/node_modules/aws-sdk/lib/sequential_executor.js:96:12)
    at IncomingMessage.onEnd (/var/config/rest/iapps/f5-service-discovery/node_modules/aws-sdk/lib/event_listeners.js:307:13)
    at emitNone (events.js:111:20)
    at IncomingMessage.emit (events.js:208:7)

**Comment by johnbairmbo** [comment]
However the same iam keys work when connected from a different system in the parent zone.

**Comment by dstokesf5** [comment]
When the BIG-IP is not in the parent zone, is it still in the same region?

**Comment by johnbairmbo** [comment]
It looks like the issue is tied to IAM rights.  What rights does this require for looking at EC2 tags?  I don't see that in the documentations.

**Comment by johnbairmbo** [comment]
I just got it working but am providing ec2:* for the rights.

**Comment by johnbairmbo** [comment]
We had another issue where the aws endpoint was not allowing local zone subnets access via it's security group.  Once we unblocked that and provided the rights in iam all is working.

**Comment by dstokesf5** [comment]
Glad to hear you were able to make progress, can this GitHub issue be closed?
